src/test/java/uk/gov/ons/census/action/model/repository/CaseRepositoryIT.java

```diff
@@ -0,0 +1,171 @@
+package uk.gov.ons.census.action.model.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.data.jpa.domain.Specification.where;
+
+import java.util.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import org.jeasy.random.EasyRandom;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.model.entity.Case;
+
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CaseRepositoryIT {
+
+  private static final String TEST_ACTION_PLAN_ID;
+  private static final Map<String, List<String>> TEST_CLASSIFIERS;
+  private EasyRandom easyRandom = new EasyRandom();
+
+  @Autowired private CaseRepository caseRepository;
+
+  static {
+    TEST_ACTION_PLAN_ID = "ddbd26c2-15af-4055-aab9-591b3735b8d3";
+
+    TEST_CLASSIFIERS = new HashMap<>();
+    TEST_CLASSIFIERS.put(
+        "treatmentCode",
+        Arrays.asList(
+            "HH_LF2R3AE",
+            "HH_LF3R3BE",
+            "HH_LF3R3AE",
+            "HH_LF2R1E",
+            "HH_LFNR2E",
+            "HH_LFNR3AE",
+            "HH_LF2R3BE",
+            "HH_LF3R2E",
+            "HH_LF2R2E",
+            "HH_LF3R1E"));
+  }
+
+  private void setupTenUnReceiptedCases() {
```

Comment:
I would prefer this to take a count as a parameter, in the calling function the expectedCount can then be passed in.  See below

Reply:
done